tests/cluster/check_vagrant.py

```diff
@@ -60,3 +63,73 @@ def check_one_host_parsing(self, monkeypatch):
         assert(node2.account.user == "vagrant")
         assert(node2.account.ssh_args.strip() == "-o 'HostName 127.0.0.2' -o 'Port 2200' -o 'UserKnownHostsFile /dev/null' -o 'StrictHostKeyChecking no' -o 'PasswordAuthentication no' -o 'IdentityFile /Users/foo/ducktape.git/.vagrant/machines/worker2/virtualbox/private_key' -o 'IdentitiesOnly yes' -o 'LogLevel FATAL'")
         assert(node2.account.ssh_hostname == '127.0.0.2')
+
+    def check_cluster_file_write(self, monkeypatch):
+        self._vagrant_ssh_data = self.two_hosts
+        monkeypatch.setattr("ducktape.cluster.vagrant.VagrantCluster._vagrant_ssh_config", lambda vc: (self._vagrant_ssh_data, None))
+        monkeypatch.setattr("ducktape.cluster.vagrant.VagrantCluster.is_aws", lambda vc: False)
+        monkeypatch.setattr("ducktape.cluster.vagrant.VagrantCluster._externally_routable_ip", lambda vc, node_account: "127.0.0.1")
+        try:
+            open(os.path.abspath(self.cluster_file))
+            assert(False, "Cluster file should not exist")
+        except IOError:
+            pass
+
+        cluster = VagrantCluster(cluster_file=self.cluster_file)
+        cluster_json_expected = {}
+        nodes = [{"hostname": node_account.hostname,
+                  "ssh_hostname": node_account.ssh_hostname,
+                  "user": node_account.user,
+                  "ssh_args": node_account.ssh_args,
+                  "externally_routable_ip": node_account.externally_routable_ip}
+                  for node_account in cluster.available_nodes]
+        cluster_json_expected["nodes"] = nodes
+        cluster_json_actual = json.load(open(os.path.abspath(self.cluster_file)))
+
+        os.remove(self.cluster_file)
+        assert(cluster_json_actual == cluster_json_expected)
+
+    def check_cluster_file_read(self, monkeypatch):
+        self._vagrant_ssh_data = self.two_hosts
+        monkeypatch.setattr("ducktape.cluster.vagrant.VagrantCluster._vagrant_ssh_config", lambda vc: (self._vagrant_ssh_data, None))
+        monkeypatch.setattr("ducktape.cluster.vagrant.VagrantCluster.is_aws", lambda vc: False)
+        monkeypatch.setattr("ducktape.cluster.vagrant.VagrantCluster._externally_routable_ip", lambda vc, node_account: "127.0.0.1")
+
+        nodes = []
+        nodes.append({
```

Comment:
Am I right in understanding that you're deliberately creating json with data that differs from the data in the mock cluster to ensure the cluster file overrides info from vagrant_ssh? It's a great idea, and I think it'd be good to call out that that is intentional in a comment.